cmd/collector/app/builder/span_handler_builder.go

```diff
@@ -170,3 +140,67 @@ func fixConfiguration(config cascfg.Configuration) cascfg.Configuration {
 	config.ProtoVersion = 4
 	return config
 }
+
+type esSpanHandlerBuilder struct {
+	logger        *zap.Logger
+	client        es.Client
+	configuration escfg.Configuration
+}
+
+func newESBuilder(config *escfg.Configuration, logger *zap.Logger) *esSpanHandlerBuilder {
+	return &esSpanHandlerBuilder{
+		configuration: *config,
+		logger:        logger,
+	}
+}
+
+func (e *esSpanHandlerBuilder) BuildHandlers() (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error) {
+	client, err := e.getClient()
+	if err != nil {
+		return nil, nil, err
+	}
+	spanStore := esSpanstore.NewSpanWriter(client, e.logger)
+
+	return buildHandlers(spanStore, e.logger, nil)
+}
+
+func buildHandlers(
+	spanStore spanstore.Writer,
+	logger *zap.Logger,
+	metricsFactory metrics.Factory,
+) (app.ZipkinSpansHandler, app.JaegerBatchesHandler, error) {
+	// TODO: currently ES has no metricsFactory, hence the workaround. Once ES adds a metricsFactory, this nullpointer check should be unnecessary.
+	var hostMetrics metrics.Factory
+	if metricsFactory != nil {
+		hostname, _ := os.Hostname()
+		hostMetrics = metricsFactory.Namespace(hostname, nil)
+	}
+
+	zSanitizer := zs.NewChainedSanitizer(
+		zs.NewSpanDurationSanitizer(logger),
+		zs.NewParentIDSanitizer(logger),
+	)
+
+	spanProcessor := app.NewSpanProcessor(
+		spanStore,
+		app.Options.ServiceMetrics(metricsFactory),
+		app.Options.HostMetrics(hostMetrics),
+		app.Options.Logger(logger),
+		app.Options.SpanFilter(defaultSpanFilter),
+		app.Options.NumWorkers(*NumWorkers),
+		app.Options.QueueSize(*QueueSize),
+	)
+
+	return app.NewZipkinSpanHandler(logger, spanProcessor, zSanitizer),
+		app.NewJaegerSpanHandler(logger, spanProcessor),
+		nil
+}
+
+func (e *esSpanHandlerBuilder) getClient() (es.Client, error) {
```

Comment:
try to have all the esSpanHanlderBuilder logic be contiguous, ie move this function further up